src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -301,6 +363,23 @@ public async Task<string> ExtractArchive(string directoryName, byte[] bytes, boo
             return GetLatestVersion(versions);
         }
 
+        /// <summary>
+        /// Check if the package exists in the respository.
+        /// </summary>
+        /// <param name="purl">The PackageURL to check.</param>
+        /// <param name="useCache">Ignored by <see cref="BaseProjectManager"/> but may be respected by inherited implementations.</param>
+        /// <returns>True if the package is confirmed to exist in the repository. False otherwise.</returns>
+        public virtual async Task<bool> PackageExists(PackageURL purl, bool useCache = true)
+        {
+            Logger.Trace("PackageExists {0}", purl?.ToString());
```

Comment:
Seeing in the log. "PackageExists {purl}" to me, would read as the package exists. However, and I could be wrong, this doesn't seem to be the case? This is just saying we are running `PackageExists` on this package.
This applies to all package managers as far as I can tell.
I would suggest something more like this, unless I am wrong and `Logger.Trace` is a way to say "we are running `PackageExists` on this package"
```suggestion
            Logger.Trace("Checking to see if {0} exists.", purl?.ToString());
```
Or something like that.

Reply:
The trace messaging is consistent with other trace calls to simply indicate entrance to the method.

End users are not expected to ever see the trace calls - this should only ever be if you are debugging the behavior and need to see which calls are being made.

Reply:
Perhaps it would be clearer to change this to say 

`Logger.Trace("PackageExists(args1,arg2")`

With the appropriate arguments.

Reply:
Thanks for the info! That's what I figured, just wasn't sure.